network/endpoint_linux.go

```diff
@@ -88,49 +94,52 @@ func (nw *network) newEndpointImpl(_ apipaClient, nl netlink.NetlinkInterface, p
 		contIfName = fmt.Sprintf("%s%s-2", hostVEthInterfacePrefix, epInfo.Id[:7])
 	}
 
-	if vlanid != 0 {
-		if nw.Mode == opModeTransparentVlan {
-			log.Printf("Transparent vlan client")
-			if _, ok := epInfo.Data[SnatBridgeIPKey]; ok {
-				nw.SnatBridgeIP = epInfo.Data[SnatBridgeIPKey].(string)
+	// epClient is non-nil only when the endpoint is created for the unit test.
+	if epClient == nil {
```

Comment:
yes. it will be nil for other cases.

Reply:
Shouldn't our test cases be changed then instead of adding code just for test cases.
We should be updating the test cases so that it mimics the actual scenarios.

Reply:
yes you are right. the problem here is the client objects are getting initialized inside this function and not outside. we can't do DI for this particular case. There is no other way i could think of. Let me know if you could think of any other way

Reply:
nope. unit tests are not added before. im adding it for 1st time for this api

Reply:
Can we add a task to fix this.